src/operation/watch/test.rs

```diff
@@ -0,0 +1,246 @@
+use std::time::Duration;
+
+use crate::{
+    bson::{doc, Document},
+    bson_util,
+    change_stream::ChangeStreamTarget,
+    cmap::{CommandResponse, StreamDescription},
+    operation::{Operation, Watch},
+    options::{
+        ChangeStreamOptions,
+        FullDocumentType,
+        ReadPreference,
+        SelectionCriteria,
+        StreamAddress,
+    },
+    test::TestClient,
+    Collection,
+    Namespace,
+};
+
+fn build_test(
+    target: impl Into<ChangeStreamTarget>,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    mut expected_body: Document,
+) {
+    let target = target.into();
+
+    let watch = Watch::new(target, pipeline, options).unwrap();
+
+    let mut cmd = watch.build(&StreamDescription::new_testing()).unwrap();
+
+    assert_eq!(cmd.name.as_str(), "aggregate");
+
+    bson_util::sort_document(&mut expected_body);
+    bson_util::sort_document(&mut cmd.body);
+
+    assert_eq!(cmd.body, expected_body);
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build() {
+    let client = TestClient::new().await;
```

Comment:
The build tests don't do anything with the database, so they shouldn't create a client and don't need to check the topology. The topology only needs to be checked the integration tests (i.e. the ones that connect to the database by calling `watch`)

Reply:
changed the ChangeStream target to not include an actual database/collection. fixed.

Reply:
This doesn't seem fixed in the latest commit.

Reply:
didn't push earlier. fixed now